src/uv.cc

```diff
@@ -90,8 +79,26 @@ void Initialize(Local<Object> target,
   UV_ERRNO_MAP(V)
 #undef V
 
-  target->Set(context, FIXED_ONE_BYTE_STRING(isolate, "errmap"),
-              err_map).FromJust();
+  args.GetReturnValue().Set(err_map);
+}
+
+void Initialize(Local<Object> target,
+                Local<Value> unused,
+                Local<Context> context,
+                void* priv) {
+  Environment* env = Environment::GetCurrent(context);
+  Isolate* isolate = env->isolate();
+  target->Set(env->context(),
+              FIXED_ONE_BYTE_STRING(isolate, "errname"),
+              env->NewFunctionTemplate(ErrName)
+                  ->GetFunction(env->context())
+                  .ToLocalChecked()).FromJust();
+
+#define V(name, _) NODE_DEFINE_CONSTANT(target, UV_##name);
```

Comment:
This is still a ton of V8 calls :)

Reply:
@addaleax I suspect this is something that is actually used by users in the wild (`util.getSystemErrorName(err)` was only introduced in v9.7.0)

I guess we can start explicitly deprecating `process.binding('uv')`? (but I am not sure how to do this since in general, `process.binding` is just deprecated)

Reply:
> I guess we can start explicitly deprecating `process.binding('uv')`? (but I am not sure how to do this since in general, `process.binding` is just deprecated)

I assume we’d want to add a greylist alongside the current whitelist for `process.binding()` for bindings that are fully runtime-deprecated?